esrally/track/params.py

```diff
@@ -580,6 +580,21 @@ def percent_completed(self):
         return self.current_bulk / self.total_bulks
 
 
+class ForceMergeParamSouce(ParamSource):
+    def __init__(self, track, params, **kwargs):
+        if len(track.indices) > 0:
+            default_index = ','.join(map(str, track.indices))
+        else:
+            default_index = "_all"
+
+        self._index_name = params.get("index", default_index)
```

Comment:
I noticed that we'd lose support for the other two parameters `max-num-segments` and `request-timeout` (see the `ForceMerge` runner class). Could you please add the following lines here in the constructor?

```python
        self._max_num_segments = params.get("max-num-segments")
        self._request_timeout = params.get("request-timeout")
```

and the `params` method needs to change to:

```python
    def params(self):
        return {
            "index": self._index_name,
            "max-num-segments": self._max_num_segments,
            "request-timeout": self._request_timeout
        }
```

you might wonder why this needs to be changed now. Previously there was a generic parameter source assigned  for the `ForceMerge` runner which would pass any parameters that are defined on the operation but now that we have a custom parameter source we need to be explicit about it.

Reply:
Thanks for explaining, makes sense. I've updated it as per above, and also added a test that validates all the params.